api/server/router/container/copy.go

```diff
@@ -98,7 +101,26 @@ func (s *containerRouter) getContainersArchive(ctx context.Context, w http.Respo
 	}
 
 	w.Header().Set("Content-Type", "application/x-tar")
-	_, err = io.Copy(w, tarArchive)
+
+	var archiveReader io.Reader
```

Comment:
I think this code should go into a function so that we'll be able to re-use it later.

```go
func writeCompressedResponse(r *http.Request,  w http.ResponseWriter, body io.Reader) error {
...
}
```